File: seed/examples/seed_example.py
"""Example of reproducible seeding in Python."""

import pathlib
import sys

import numpy as np
import torch


CURRENT_DIR = pathlib.Path(__file__).resolve().parent
CODE_DIR = CURRENT_DIR.parent
if str(CODE_DIR) not in sys.path:
    sys.path.insert(0, str(CODE_DIR))

from seed import seed_everything


def main():
    seed_everything(123)

    print("python_random_numpy:", np.random.rand(3).tolist())
    print("python_random_torch:", torch.rand(3).tolist())


if __name__ == "__main__":
    main()

File: seed/examples/seed_example.R
library(reticulate)

seed_file <- normalizePath(file.path("code", "seed.py"), winslash = "/", mustWork = TRUE)
python_bin <- Sys.which("python")

if (python_bin == "") {
  stop("Python interpreter not found in PATH.")
}

set.seed(123)
use_python(python_bin, required = TRUE)
source_python(seed_file)
seed_everything(123)
np <- import("numpy")

cat("R random:", paste(round(rnorm(3), 6), collapse = ", "), "\n")
cat("Python NumPy random:", paste(round(as.numeric(np$random$rand(3L)), 6), collapse = ", "), "\n")
